src/ophys_etl/transforms/classification.py

```diff
@@ -0,0 +1,304 @@
+        }] * len(roi_data)
+    rois = [r["coo_roi"] for r in roi_data]
+    traces = []
+    np_traces = []
+
+    traces_file = h5py.File(parser.args["traces_path"], "r")
+    np_traces_file = h5py.File(parser.args["neuropil_traces_path"], "r")
+    traces_data = traces_file[parser.args["traces_data_key"]]
+    np_traces_data = np_traces_file[parser.args["neuropil_traces_data_key"]]
+    for n in range(len(roi_data)):
+        # Downsample traces by accessing on axis that the h5 file should be
+        # more performant on
+        trace = downsample(
+            traces_data[n, :], parser.args["trace_sampling_fps"],
+            parser.args["downsample_to"])
+        np_trace = downsample(
+            np_traces_data[n, :], parser.args["trace_sampling_fps"],
+            parser.args["downsample_to"])
+        traces.append(trace)
+        np_traces.append(np_trace)
+    traces_file.close()
+    np_traces_file.close()
+    return rois, metadata, traces, np_traces
+
+
+def downsample(trace: np.ndarray, input_fps: int, output_fps: int):
+    """Downsample 1d array using an averaging strategy.
+    If a full bin cannot be populated, the data will be truncated
+    to approximate required output_fps.
+    Example: downsampling from 5 fps to 2 fps
+    for a trace with 13 points would need to drop the last data point.
+
+    Parameters
+    ----------
+    trace: np.ndarray
+        1d array of values to downsample
+    input_fps: int
+        The FPS that the trace data was captured at
+    output_fps: int
+        The desired FPS of the trace
+    Returns
+    -------
+    np.ndarray
+        1d array of values, downsampled to output_fps (may be approximate)
+    """
+    if input_fps == output_fps:
+        return trace
+    points = trace.shape[0]
+    output_size = math.floor(points * output_fps / input_fps)
+    bin_size = math.floor(points / output_size)
+    if points / output_size != bin_size:    # if the bin needed to be rounded
+        warnings.warn(f"Can't evenly downsample {points} points at "
+                      f"{input_fps} FPS to {output_fps} FPS. Output fps is "
+                      "approximate.")
+    truncate = points % output_size
+    if truncate != 0:
+        downsampled = (trace[:-truncate].reshape(output_size, bin_size)
+                       .mean(axis=1))
+    else:
+        downsampled = trace.reshape(output_size, bin_size).mean(axis=1)
+    return downsampled
+
+
+def main(parser):
+    with open(parser.args["roi_masks_path"], "r") as f:
+        raw_roi_data = json.load(f)
+    roi_input_schema = RoiJsonSchema(many=True)
+    roi_data = roi_input_schema.load(raw_roi_data)
+    rois, metadata, traces, _ = _munge_data(parser, roi_data)
+    # TODO: add neuropil traces later
+    features = FeatureExtractor(rois, traces, metadata).run()
+    model = joblib.load(parser.args["classifier_model_path"])
+    predictions = model.predict(features)
+    if len(predictions) != len(roi_data):
```

Comment:
how would this situation arise?

Reply:
It likely wouldn't, but I am not ruling out that something weird could happen in FeatureExtractor. The reason I did this is because the `zip` method below will not raise an error if the arrays are not the same length, but will truncate at the end of the shortest iterable.

Reply:
ok